### PR TITLE
Add some functions and improvements in framework for etcd related cases

### PR DESCRIPTION
**Pull request by ypu**
This include some new function, improvement and bug fix in framework. The new functions and improvement are needed by etcd test cases.

### CODE REVIEW

File: environment.py [hunk 1]
```diff
@@ -137,13 +142,23 @@ def remote_cmd(cmd, host=None, ignore_rc=False, **kwargs):
...
+            ).run_async(context.test_timeout)
+            return poller
+        else:
+            context.result = ansible.runner.Runner(
+                     module_name=cmd,
+                     inventory=inventory,
+                     pattern=host,
+                     #remote_user=remote_user,
+                     **kwargs
+            ).run()
```
Comment (miabbott) [inline environment.py]: nit:  The indents in these lines look off.  Try using `flake8` or `pep8` to detect style issues.

**Comment by miabbott** [comment]
@ypu LGTM other than the one style comment.  

@chuanchang could you have a look, too?

File: environment.py (hunk 1 above)
Comment (ypu) [inline environment.py]: pep8 will not raise warning for these lines. But it is OK for me to update and during my check there are several style problem exist in the code too. Will format a patch for it. Thanks for reminding.

**Comment by chuanchang** [comment]
LGTM.

**Comment by miabbott** [comment]
Thanks for the style fixes @ypu.  And thanks for the extra look @chuanchang.

**Pull request merged**